crates/trigger/src/cli.rs

```diff
@@ -114,21 +117,43 @@ where
         let working_dir = std::env::var(SPIN_WORKING_DIR).context(SPIN_WORKING_DIR)?;
         let locked_url = std::env::var(SPIN_LOCKED_URL).context(SPIN_LOCKED_URL)?;
 
-        let loader = TriggerLoader::new(working_dir, self.allow_transient_write);
+        // TODO: I assume there is a way to do this with a single let mut loader: Box<dyn Loader>
+        // variable instead of the entire executor.
```

Comment:
Would be good to dedupe for sure... `fn xxx_impl(loader: impl Loader, ...) -> Executor` then call it with a TriggerLoader or OciTriggerLoader depending on the value of `self.oci`?